import "libnutrition/proto/data/meal.proto";
import "libnutrition/proto/service/storage_modification.proto";
import "libnutrition/proto/service/error.proto";

message MealIdentifier 
{
	optional int32 userId = 1;
	optional string date_ISO8601 = 2;
	optional int32 mealId = 3;
}

message MealLoadRequest 
{
	optional bool all = 1 [default = false];
	optional bool omit = 2 [default = false];
	optional bool nameAndIdOnly = 3 [default = false];
	optional bool omitGenerics = 4 [default = false];
		
	repeated MealIdentifier requestedIds = 5;	
}

message MealLoadResponse
{
	repeated MealData meals = 1;
	optional Error error = 100;
}

message MealStoreRequest
{
	repeated MealData meals = 1;
}

message MealStoreResponse
{
	repeated MealIdentifier storedIds = 1;
	
	message MealComponentModification {
		optional MealIdentifier collectionId = 1; 
		repeated StorageModification componentModifications = 2;
	}
	
	repeated MealComponentModification componentModifications = 2;
	
	optional Error error = 100;
}

message MealDeleteRequest
{
	repeated string deleteDates_ISO8601 = 1;
	repeated MealIdentifier deleteIds = 2;
}

message MealDeleteResponse
{
	repeated MealIdentifier deletedIds = 1;
	optional Error error = 100;
}